Avoid hard coding of version info in configure.ac

pete4abw opened the pull request:
Instead, the shell program, fvwm-version-str.sh will use git describe
and git show to obtain tag and release information from the log.

If the annotated release tag is at the HEAD, the version will be
considered a release. If there are commits after the release tag,
the micro version will have the tag offset as well as the commit
of the HEAD appended to it. Going forward, versions will appear as:

Fvwm3 Configuration:

  Version:     1.0.3 (released)

  Executables: /usr/local/bin
  Man pages:   /usr/local/share/man
  Modules:     /usr/local/libexec/fvwm3/1.0.3

for a Release, and

Fvwm3 Configuration:

  Version:     1.0.2-25-b9435cd (not released)

  Executables: /usr/local/bin
  Man pages:   /usr/local/share/man
  Modules:     /usr/local/libexec/fvwm3/1.0.2-25-b9435cd

when HEAD commit <> release tag commit. This example indicated there
are 25 commits after the last release.

The program fvwm-version-str.sh can be run independently to test
the impact of different scenarios. It is fuly documented and may be
altered to taste.

The file, configure.ac currently searches for annotated tags, not
lightweight tags. Removing the option -r from the m4_esyscmd_s command
will allow it to search for the latest tag.

m4_define([v_maj],		[m4_esyscmd_s([./utils/fvwm-version-str.sh major -r])])

On branch ph/gitdesc
	modified:   Makefile.am
	modified:   configure.ac
	modified:   utils/fvwm-version-str.sh

* **What does this PR do?**

* **Screenshots (if applicable)**

* **PR acceptance criteria** (reminder only, please delete once read)

  - Your commit message(s) are descriptive.  See:

    https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

  - [https://raw.githubusercontent.com/fvwmorg/fvwm3/master/doc/README]Documentation updated (where appropriate)

  - Style guide followed (try and match the surrounding code where possible)

  - All tests are passing:  although this is automatic, Codacy will often
    highlight additional considerations which will need to be addressed before
    the PR can be merged.

* **Issue number(s)**

If this PR addresses any issues, please ensure the appropriate commit
message(s) contains:

```
Fixes #XXX
```

at the end of your commit message, where `XXX` should be replaced with the
relevant issue number.

If there is more than one issue fixed then use:

```
Fixes #XXX, fixes #YYY, fixes #ZZZ
```

ThomasAdam commented on utils/fvwm-version-str.sh:
```diff
@@ -10,12 +10,180 @@
+	major=$(echo $version | cut -d. -f1)
+}
+
+[ ! $(which git) ] && die "Something very wrong: git not found."
```
```
type -p git >/dev/null || die "....."
```

ThomasAdam commented on utils/fvwm-version-str.sh:
```diff
@@ -10,12 +10,180 @@
+	describe_tag=$(git describe $tagopt --long --abbrev=$commit_length)
+
+	# git describe prefixes commit with the letter `g'
+	describe_tag=${describe_tag/g/}
```
Ah.  Please don't do that.  The "g-" prefix is useful and I use it all the time to run `git checkout` on it.  Removing this "g" breaks that.  It's actually there for a reason.  From `git-describe(1)`:

```
       The number of additional commits is the number of commits which would be displayed by "git log v1.0.4..parent". The hash suffix is "-g" + unambiguous abbreviation for the tip commit of parent
       (which was 2414721b194453f058079d897d13c4e377f92dc6). The "g" prefix stands for "git" and is used to allow describing the version of a software depending on the SCM the software is managed with.
       This is useful in an environment where people may use different SCMs.
```

ThomasAdam reviewed (changes requested):
Hi @pete4abw 

Thanks for sending this over.  The PR itself is still marked as in-progress, so am not sure if you're expecting me to fully review it yet, but I've left some comments for you nonetheless.

It currently fails to build -- you might want to check the to see why.

pete4abw commented:
**Thank you** for the review, @ThomasAdam . This is what I needed since I am new in the project.

I added quoting for VERSIONINFO in configure.ac Compiles now
Removed licensing for fvwm_version_str.sh. Add whatever you need.
Switched from using `which` to `type -p`
Restored leading `g` to commit value

I was confused by the AC_DEFINE_UNQUOTED. It messed up the created `config.h` file where
**VERSIONINFO**=not released instead of "not released"
with no quotes.

Have another looksee!
